src/main/java/relay/data_access/FirebaseCourseDataAccessObject.java

```diff
@@ -43,39 +48,67 @@ public void createCourse(Course course) throws InterruptedException, ExecutionEx
      * @return An ArrayList of Course objects associated with the specified instructor.
      * @throws RuntimeException If an InterruptedException or ExecutionException occurs during Firestore operations.
      */
-    public ArrayList<Course> getCoursesByInstructor(Instructor instructor) throws InterruptedException, ExecutionException {
+    public ArrayList<Course> getCoursesByInstructor(Instructor instructor) {
         Query query = db.collection("courses")
                 .whereEqualTo("instructorID", instructor.getInstructorID());
 
         ApiFuture<QuerySnapshot> querySnapshot = query.get();
 
         ArrayList<Course> courses = new ArrayList<>();
-        for (DocumentSnapshot document : querySnapshot.get().getDocuments()) {
+        try {
+            for (DocumentSnapshot document : querySnapshot.get().getDocuments()) {
 
-            Map<String, Object> courseData = document.getData();
+                Map<String, Object> courseData = document.getData();
 
-            String courseID = document.getId();
-            String courseName = (String) courseData.get("courseName");
+                String courseID = document.getId();
+                String courseName = (String) courseData.get("courseName");
 
-            Course course = new Course(courseID, courseName, instructor);
-            courses.add(course);
+                Course course = new Course(courseName, instructor);
+                course.setCourseID(courseID);
+                courses.add(course);
 
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException(e);
         }
         return courses;
     }
 
     /**
-     * Checks if a course with the given courseID exists in the Firestore database.
+     * Checks if a course with the given course ID exists in the Firestore database.
      *
-     * @param courseID The ID of the course to check for existence.
-     * @return true if the course exists, false otherwise.
-     * @throws RuntimeException If an ExecutionException or InterruptedException occurs during Firestore operations.
+     * @param courseID the ID of the course to check for existence
+     * @return {@code true} if the course with the specified ID exists, {@code false} otherwise
      */
-    public boolean exists(String courseID) throws InterruptedException, ExecutionException {
+    public boolean exists(String courseID) {
         ApiFuture<DocumentSnapshot> retrievedcourseDocument = db.collection("courses").document(courseID).get();
-        DocumentSnapshot courseDocument = retrievedcourseDocument.get();
+        DocumentSnapshot courseDocument = null;
+        try {
+            courseDocument = retrievedcourseDocument.get();
```

Comment:
You should just be able to do `retrievedcourseDocument.get()`, why make it null first?